sound/soc/sof/intel/hda.c

```diff
@@ -692,14 +692,10 @@ static int hda_generic_machine_select(struct snd_sof_dev *sdev)
 		}
 
 		/*
-		 * If no machine driver is found, then:
-		 *
-		 * hda machine driver is used if :
-		 * 1. there is one HDMI codec and one external HDAudio codec
-		 * 2. only HDMI codec
+		 * If no machine driver is found, use the generic
+		 * hda machine driver if at least the HDMI codec is present
 		 */
-		if (!pdata->machine && codec_num <= 2 &&
-		    HDA_IDISP_CODEC(bus->codec_mask)) {
```

Comment:
@kv2019i will this case ever be true that codec_num will be > 2? In the previous patch, you'd have adjusted the codec_mask to drop the codec that failed to probe isnt it?

Reply:
@ranj063 There are two different scenarios tackled. First is that we have a faulty external codec. In this case, we may go from iDisp+Ext=>IDisp_only, or from IDisp+ext+ext2=>iDisp+ext1. 

Second case we have a system that has multiple working external HDA codecs attached on the bus. In this case the probes will succeed for all codecs, but we do not have a machine driver (and related infra) to tackle this configuration. What we can do is to still enable at least the first external HDA codec.

But, but, while responding to @plbossart previous comment, I realized guaranteeing we can pick configure the right codec, might be more complex than what I have here. If I can't solve that, then I have to revert to just address the first scenario (i.e. faulty codec probes).

Reply:
> Second case we have a system that has multiple working external HDA codecs attached on the bus. In this case the probes will succeed for all codecs, but we do not have a machine driver (and related infra) to tackle this configuration. What we can do is to still enable at least the first external HDA codec.

@kv2019i this would be a flawed strategy isnt it? If there is a working codec, we need to provide the right machine driver that manages it?

Reply:
@ranj063 Well, the generic HDA driver is a fallback in any case. Yes we should have a more specific machine driver, but if we reach this point in code, search for that machine driver already failed. So the question is then that should we at least enable HDMI audio, and maybe a single HDA codec , which we do have machine driver, or leave the end user with no audio at all.

But yeah, not so clear cut. Currently I anyways have pieces missing to tackle this, so I need to do some more homework for this.

Reply:
@kv2019i with my multi-client stuff, I do plan to separate out the HDMI codec into a separate client. With that change, you'd probably have the fallback with just the HDMI codecs working if they are probed successfully.